src/main/java/com/sda/calc3/NumberWithTextRepresentation.java

```diff
@@ -0,0 +1,40 @@
+package com.sda.calc3;
+
+public class NumberWithTextRepresentation <T,String> extends java.lang.Number {
```

Comment:
Hmm, już prawie dobrze, tylko nie do końca chodziło mi o to, by klasa dziedziczyła po `Number`. A może tak niech T dziedziczy po `Number`? Wtedy mogłabyś uniknąć w getterach rzutowania na `int`, `long`... Myślę, że dzięki zastosowaniu tego mechanizmu kod byłby nieco prostszy.